### ISSUE TITLE
marshalparser error with Python 3.14

### ISSUE THREAD

**Issue by befeleme** [fedora-python/marshalparser#34]
The marshalparser tests failed in https://src.fedoraproject.org/rpms/python3.14/pull-request/4

```
+ marshalparser /usr/lib64/python3.14/test/__pycache__/test_compileall.cpython-314.opt-2.pyc
Cannot read/parse byte 24 b'\x18' on possition 0
Might be error or unsupported TYPE
```

**Comment by hroncok** [comment]
Cross-referencing https://github.com/keszybz/add-determinism/issues/42

**Comment by frenzymadness** [comment]
Thanks for the report. I'm pretty sure the problem is caused by the new type in marshal format: https://github.com/python/cpython/commit/c6127af8685c2a9b416207e46089cee79d028b85#diff-46d545a32ea7176926cf1a963f25cdac2cf1c8098b3e020ccc77b63b2149d760

I'll use the python3.14 package for testing and a fix should be ready soon.

**Comment by hroncok** [comment]
https://github.com/keszybz/add-determinism/commit/b59c4fc739a37f74d92dd765c203ef1bbe8db822 also adds support for *slice*

**Comment by frenzymadness** [comment]
Fixed in 0.4.0
